### DISCUSSION TITLE
What makes langcc-based parsers Clang-dependent?

### DISCUSSION THREAD

**Discussion by omasanori** [jzimmerman/langcc#45] [Q&A]
Hi,
I am trying langcc for my next hobby project and wonder if the condition below, especially Clang, is mandatory or not.

> langcc outputs code intended to be compiled with clang++ using the C++'17 standard

**Comment by misutoneko** [comment]
Maybe you'll find [my little wrapper script](https://gist.github.com/misutoneko/34d3d9478036773a24ae7ba6386ec211) interesting.
I often found myself wondering "which C++ standard is used in this repository?".
But I had no idea how to get a quick answer other than reading tons of code.
So when I stumbled upon tree-sitter I knew I had a real nice use case for it :D

The script doesn't handle all that many C++17 features, but:
C++17 does seem to be a hard requirement, since nested namespaces and C++17 headers are present.

As for Clang, I have no clue. It looks like it's needed for compiling langcc itself though.
